client/src/components/Upload/UploadModal.vue

```diff
@@ -12,230 +17,56 @@
         :id="id"
     >
         <template v-slot:modal-header>
-            <h4 class="title" tabindex="0">{{ title }}</h4>
+            <h4 class="title" tabindex="0">{{ title | localize }}</h4>
         </template>
-        <b-tabs v-if="ready">
-            <b-tab title="Regular" id="regular" button-id="tab-title-link-regular" v-if="showRegular">
-                <default :app="this" :lazy-load-max="50" :multiple="multiple" :selectable="callback != null" />
-            </b-tab>
-            <b-tab title="Composite" id="composite" button-id="tab-title-link-composite" v-if="showComposite">
-                <composite :app="this" />
-            </b-tab>
-            <b-tab title="Collection" id="collection" button-id="tab-title-link-collection" v-if="showCollection">
-                <collection :app="this" />
-            </b-tab>
-            <b-tab title="Rule-based" id="rule-based" button-id="tab-title-link-rule-based" v-if="showRules">
-                <rules-input :app="this" />
-            </b-tab>
-        </b-tabs>
-        <div v-else>
-            <loading-span message="Loading required information from Galaxy server." />
-        </div>
+
+        <CurrentUser v-slot="{ user }">
+            <UserHistories v-if="user && user.id" v-slot="{ currentHistoryId }">
```

Comment:
This is nice! Can we do this outside of vue as well or can we wrap backbone components in vue components? I don't expect we'll be able to modernize the tool form before the release ...

Reply:
@mvdbeek 
Hey I'm actually doing the upload stuff in a complete replacement of the Upload Modal. I'm going to close this PR and open 2 others.

As for "outside vue", one thing you could always do is go straight to the store to get config data you need, then provide that data as props to the Vue component you mount from backbone, if for intance, you're not able to modernize the tool-form.

Alternatively you could just redefine what you mount from Backbone as being a component that leverages these wrappers. I can show you how to pass attributes and listeners from a parent to a child to make it all completely transparent.

Reply:
This PR was starting to balloon into a bigger one than I wanted.

Check this one out for addressing the bugs you originally pinged me for:
https://github.com/galaxyproject/galaxy/pull/11172